test/extended/cluster/utils.go

```diff
@@ -0,0 +1,347 @@
+		}
+	}
+	return configMapName
+}
+
+// getClusterData will return map containing updated strings based on custer data
+func getClusterData(c *kclientset.Clientset, config map[string]string) map[string]string {
+	newConfig := make(map[string]string)
+	for k, v := range config {
+		if k == "routerIP" {
+			// TODO sjug: make localhost func
+			//v = localhost(f)
+			v = "127.0.0.1"
+		} else if k == "targetHost" {
+			// getEndpointsWithLabel will not return single string
+			v = concatenateIP(getEndpointsWithLabel(c, config["match"]))
+		}
+		newConfig[k] = v
+	}
+	return newConfig
+}
+
+func concatenateIP(endpointInfo []ServiceInfo) (ip string) {
+	for i := range endpointInfo {
+		ip += endpointInfo[i].IP + ","
+	}
+	return
+}
+
+func getEndpointsWithLabel(c *kclientset.Clientset, label string) (endpointInfo []ServiceInfo) {
+	selector, _ := labels.Parse(label)
+	endpoints, err := c.Core().Endpoints("").List(kapi.ListOptions{LabelSelector: selector})
+	if err != nil {
+		panic(err.Error())
+	}
+	for _, v := range endpoints.Items {
+		if len(v.Subsets) > 0 {
+			for _, ep := range v.Subsets[0].Addresses {
+				end := ServiceInfo{v.ObjectMeta.Name, ep.IP, v.Subsets[0].Ports[0].Port}
+				fmt.Printf("For endpoint \"%s\", the IP is %v, the port is %d\n", end.Name, end.IP, end.Port)
+				endpointInfo = append(endpointInfo, end)
+			}
+		}
+	}
+
+	return
+}
+
+//func getPodDetailsWithLabel(f *framework.Framework, label string) (podInfo []ServiceInfo) {
+//	selector := v1.ListOptions{LabelSelector: label}
+//	pods, err := f.ClientSet.Core().Pods("").List(selector)
+//	if err != nil {
+//		panic(err.Error())
+//	}
+//	for _, v := range pods.Items {
+//		pod, err := f.ClientSet.Core().Pods(v.ObjectMeta.Namespace).Get(v.ObjectMeta.Name, metav1.GetOptions{})
+//		if err != nil {
+//			panic(err.Error())
+//		}
+//		info := ServiceInfo{pod.Name, pod.Status.PodIP, pod.Spec.Containers[0].Ports[0].ContainerPort}
+//		fmt.Printf("For pod \"%s\", the IP is %v, the port is %d\n", info.Name, info.IP, info.Port)
+//		podInfo = append(podInfo, info)
+//	}
+//
+//	return
+//}
+
+func newConfigMap(ns string, name string, vars map[string]string) *kapi.ConfigMap {
+	return &kapi.ConfigMap{
+		ObjectMeta: kapi.ObjectMeta{
+			Namespace: ns,
+			Name:      name,
+		},
+		Data: vars,
+	}
+}
+
+// CreateTemplate does regex substitution against the template file, then creates the template
+func CreateTemplate(oc *exutil.CLI, baseName string, ns string, configPath string, numObjects int, tuning *TuningSetType) {
+	// Try to read the file
+	content, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		framework.Failf("Error reading file: %s", err)
+	}
+
+	// ${IDENTIFER} is what we're replacing in the file
+	regex, err := regexp.Compile("\\${IDENTIFIER}")
+	if err != nil {
+		framework.Failf("Error compiling regex: %v", err)
+	}
+
+	for i := 0; i < numObjects; i++ {
+		result := regex.ReplaceAll(content, []byte(strconv.Itoa(i)))
```

Comment:
This was all written for kubernetes, specifically to try and process duplicate named objects. This is not used for OpenShift.